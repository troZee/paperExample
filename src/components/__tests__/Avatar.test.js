import * as React from 'react';
import renderer from 'react-test-renderer';
import * as Avatar from '../Avatar/Avatar.tsx';

it('renders avatar with text', () => {
  const tree = renderer.create(<Avatar.Text label="XD" />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders avatar with text and custom size', () => {
  const tree = renderer.create(<Avatar.Text size={96} label="XD" />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders avatar with text and custom background color', () => {
  const tree = renderer
    .create(<Avatar.Text style={{ backgroundColor: '#FF0000' }} label="XD" />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders avatar with text and custom colors', () => {
  const tree = renderer
    .create(<Avatar.Text color="#FFFFFF" label="XD" />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders avatar with icon', () => {
  const tree = renderer.create(<Avatar.Icon icon="information" />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders avatar with icon and custom background color', () => {
  const tree = renderer
    .create(
      <Avatar.Icon style={{ backgroundColor: '#FF0000' }} icon="information" />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders avatar with image', () => {
  const tree = renderer
    .create(<Avatar.Image source={{ src: 'avatar.png' }} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
