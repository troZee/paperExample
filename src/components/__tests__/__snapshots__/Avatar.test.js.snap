// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders avatar with icon 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#6200ee",
        "borderRadius": 32,
        "height": 64,
        "width": 64,
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      Object {},
    ]
  }
>
  <Text
    accessibilityElementsHidden={true}
    allowFontScaling={false}
    importantForAccessibility="no-hide-descendants"
    pointerEvents="none"
    style={
      Array [
        Object {
          "color": "#ffffff",
          "fontSize": 38.4,
        },
        Array [
          Object {
            "transform": Array [
              Object {
                "scaleX": 1,
              },
            ],
          },
          Object {
            "backgroundColor": "transparent",
          },
        ],
        Object {
          "fontFamily": "Material Design Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`renders avatar with icon and custom background color 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FF0000",
        "borderRadius": 32,
        "height": 64,
        "width": 64,
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      Object {},
    ]
  }
>
  <Text
    accessibilityElementsHidden={true}
    allowFontScaling={false}
    importantForAccessibility="no-hide-descendants"
    pointerEvents="none"
    style={
      Array [
        Object {
          "color": "#ffffff",
          "fontSize": 38.4,
        },
        Array [
          Object {
            "transform": Array [
              Object {
                "scaleX": 1,
              },
            ],
          },
          Object {
            "backgroundColor": "transparent",
          },
        ],
        Object {
          "fontFamily": "Material Design Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`renders avatar with image 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#6200ee",
        "borderRadius": 32,
        "height": 64,
        "width": 64,
      },
      undefined,
    ]
  }
>
  <Image
    source={
      Object {
        "src": "avatar.png",
      }
    }
    style={
      Object {
        "borderRadius": 32,
        "height": 64,
        "width": 64,
      }
    }
  />
</View>
`;

exports[`renders avatar with text 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#6200ee",
        "borderRadius": 32,
        "height": 64,
        "width": 64,
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      Object {},
    ]
  }
>
  <Text
    numberOfLines={1}
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "System",
          "fontWeight": "400",
          "textAlign": "left",
        },
        Array [
          Object {
            "textAlign": "center",
            "textAlignVertical": "center",
          },
          Object {
            "color": "#ffffff",
            "fontSize": 32,
            "lineHeight": 64,
          },
          undefined,
        ],
      ]
    }
  >
    XD
  </Text>
</View>
`;

exports[`renders avatar with text and custom background color 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FF0000",
        "borderRadius": 32,
        "height": 64,
        "width": 64,
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      Object {},
    ]
  }
>
  <Text
    numberOfLines={1}
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "System",
          "fontWeight": "400",
          "textAlign": "left",
        },
        Array [
          Object {
            "textAlign": "center",
            "textAlignVertical": "center",
          },
          Object {
            "color": "#ffffff",
            "fontSize": 32,
            "lineHeight": 64,
          },
          undefined,
        ],
      ]
    }
  >
    XD
  </Text>
</View>
`;

exports[`renders avatar with text and custom colors 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#6200ee",
        "borderRadius": 32,
        "height": 64,
        "width": 64,
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      Object {},
    ]
  }
>
  <Text
    numberOfLines={1}
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "System",
          "fontWeight": "400",
          "textAlign": "left",
        },
        Array [
          Object {
            "textAlign": "center",
            "textAlignVertical": "center",
          },
          Object {
            "color": "#FFFFFF",
            "fontSize": 32,
            "lineHeight": 64,
          },
          undefined,
        ],
      ]
    }
  >
    XD
  </Text>
</View>
`;

exports[`renders avatar with text and custom size 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#6200ee",
        "borderRadius": 48,
        "height": 96,
        "width": 96,
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      Object {},
    ]
  }
>
  <Text
    numberOfLines={1}
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "System",
          "fontWeight": "400",
          "textAlign": "left",
        },
        Array [
          Object {
            "textAlign": "center",
            "textAlignVertical": "center",
          },
          Object {
            "color": "#ffffff",
            "fontSize": 48,
            "lineHeight": 96,
          },
          undefined,
        ],
      ]
    }
  >
    XD
  </Text>
</View>
`;
